src/gsy_e/models/market/future.py

```diff
@@ -178,33 +178,44 @@ def _expire_orders(self, orders: "FutureOrders", current_market_time_slot: DateT
             if time_slot <= current_market_time_slot:
                 del orders.slot_order_mapping[time_slot]
 
-    def delete_orders_in_old_future_markets(self, current_market_time_slot: DateTime) -> None:
+    def delete_orders_in_old_future_markets(self, last_slot_to_be_deleted: DateTime
+                                            ) -> None:
         """Delete order and trade buffers."""
-        self._expire_orders(self.offers, current_market_time_slot)
-        self._expire_orders(self.bids, current_market_time_slot)
+        self._expire_orders(self.offers, last_slot_to_be_deleted)
+        self._expire_orders(self.bids, last_slot_to_be_deleted)
 
         self.offer_history = self._remove_old_orders_from_list(
-            self.offer_history, current_market_time_slot)
+            self.offer_history, last_slot_to_be_deleted)
         self.bid_history = self._remove_old_orders_from_list(
-            self.bid_history, current_market_time_slot)
+            self.bid_history, last_slot_to_be_deleted)
         self.trades = self._remove_old_orders_from_list(
-            self.trades, current_market_time_slot)
+            self.trades, last_slot_to_be_deleted)
 
-    def create_future_markets(self, current_market_time_slot: DateTime,
-                              slot_length: duration,
-                              config: "SimulationConfig") -> None:
-        """Add sub dicts in order dictionaries for future market slots."""
-        if not GlobalConfig.FUTURE_MARKET_DURATION_HOURS:
-            return
-        future_time_slot = current_market_time_slot.add(minutes=slot_length.total_minutes())
-        most_future_slot = (current_market_time_slot +
-                            duration(hours=GlobalConfig.FUTURE_MARKET_DURATION_HOURS))
-        while future_time_slot <= most_future_slot:
+    @staticmethod
+    def _get_market_slot_duration(_current_time: DateTime, config: "SimulationConfig") -> duration:
+        return config.slot_length
+
+    def _create_future_market_slots(
+            self, start_time: DateTime, end_time: DateTime, config: "SimulationConfig") -> None:
+        future_time_slot = start_time
+        while future_time_slot <= end_time:
             if (future_time_slot not in self.slot_bid_mapping and
                     is_time_slot_in_simulation_duration(future_time_slot, config)):
```

Comment:
At the moment, it looks like this check never returns `True`. This is the method in gsy-framework: https://github.com/gridsingularity/gsy-framework/blob/bd4851c95241904e2c30788dbb0e34e78903dd2e/gsy_framework/utils.py#L552-L570

For our simulation, this check returns `False` because `config.end_date` is just 1 day in the future (it uses the default `GlobalConfig.sim_duration` value). So the following check is always false:
```py
start_date <= time_slot < end_date
```